src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/DocumentProcessor.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.ScannableItem;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.ScannableItemRepository;
+import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.DocumentNotFoundException;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.document.DocumentManagementService;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.document.output.Pdf;
+
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class DocumentProcessor {
+
+    private final DocumentManagementService documentManagementService;
+    private final ScannableItemRepository scannableItemRepository;
+
+    private static final Logger log = LoggerFactory.getLogger(DocumentProcessor.class);
+
+    public DocumentProcessor(
+        DocumentManagementService documentManagementService,
+        ScannableItemRepository scannableItemRepository
+    ) {
+        this.documentManagementService = documentManagementService;
+        this.scannableItemRepository = scannableItemRepository;
+    }
+
+    @Transactional
+    public void processPdfFiles(List<Pdf> pdfs, List<ScannableItem> scannedItems) {
+        // TODO check scannedItems.size == pdfFiles.size
```

Comment:
Should we care when such thing happen? Or should we put an assertion on that?

Reply:
Yes we should but not part of this ticket.